pkg/controller/servicebindingrequest/reconciler.go

```diff
@@ -200,6 +200,15 @@ func (r *reconciler) Reconcile(request reconcile.Request) (reconcile.Result, err
 		logger.Error(err, "Error add watching application GVR")
 	}
 
+	if sbr.GetDeletionTimestamp() != nil && sbr.GetOwnerReferences() != nil {
+		logger := logger.WithName("Deleting SBR when it has ownerReference")
+		logger.Debug("Removing resource finalizers...")
+		sbr.SetFinalizers(removeStringSlice(sbr.GetFinalizers(), finalizer))
```

Comment:
#498 introduces a function to remove finalizers, perhaps cherry-picking the commit if possible will make life easier.

Reply:
> Other than the review comments on the code, I'd like to understand more the rationale of this change; it seems that the unbind process is not executed before removing the finalizer and updating the SBR.
> 
> Perhaps the feature scenario here was related to an application owning the SBR, but this is not the only ownership scenario one might have as any other arbitrary resource can be used as resource reference and those should be taken in consideration.

@isutton , why unbind particularly failed here is because unbind() https://github.com/redhat-developer/service-binding-operator/blob/master/pkg/controller/servicebindingrequest/binder.go#L578 was trying to find objects and the app is deleted previously and provides a deletionTimeStamp on SBR as well.  For any other application issue, it is tackled here https://github.com/redhat-developer/service-binding-operator/pull/498/files#diff-fae660884df4e35af60a04c5ddf930daR291 but in this case, this app is not owning the sbr. So I can mark this PR as dependent on #498 . But we do need a condition of sbr.GetDeletionTimestamp() != nil && sbr.GetOwnerReferences() when the app is owning the sbr

Reply:
As far as #542  is concerned, it isn't really a feature request. If we want a bigger refactor of unbind and tackling various scenarios, which is out of the scope of #542  then we should maybe have a Cabal discussion for the same, and have a separate refactor for unbind story. For now ODO folks are blocked on this.

Reply:
+1 . This feels like a bug because we are potentially messing with the usual `ownerReference` expectations otherwise ?

Reply:
@isutton  I have reused the function added by #498